### PR TITLE
Task/banner component

### PR DESCRIPTION
**Pull request by gomezjuliana**
Hey all, 

This is the first steps of the banner component needed in the AppShell 😀I wanted to get this first very out quickly to get feedback on the structure. How does this feel to you? Any details I might be over looking? Any comments I should add in? Thanks in advance!

Figma files: https://www.figma.com/file/WLpZwrUBDtLmF9lilQ7BgZ7H/BDS-Navbar?node-id=186%3A254

### CODE REVIEW

**Comment by gomezjuliana** [comment]
For reference, this is how I'm using it in Reply for the impersonation banner:

```<InboxWrapper>
          <AppShell
            activeProduct="reply"
            user={{
              name: currentUser.name,
              email: this.props.userAccountSettings.email,
              avatar: profileImageUrl,
              menuItems: userMenuItems,
            }}
            helpMenuItems={helpMenuOptions}
            sidebar={this.renderSideBar()}
            content={this.renderActiveContainer()}
            bannerOptions={this.renderBanner()}
          />
```

```
renderBanner = () => {
    if (localStorage.get('user')) {
      const user = localStorage.get('user');
      if (user.impersonate) {
        return {
          text: 'hey there, this is ajhvdjah a banner',
          actionButton: {
            label: 'click me for an action',
            action: () => {
              console.log('you clicked me!');
            },
          },
        };
      }
      return null;
    }
    return null;
  };
```

File: src/components/Banner/Banner.jsx [hunk 1]
```diff
@@ -19,14 +19,22 @@ export default class Banner extends React.Component {
...
+                {this.props.text}
+              </Text>
+              <Button
+                type="primary"
+                onClick={this.props.actionButton.action}
+                label={this.props.actionButton.label}
+              />
+            </React.Fragment>
+          )}
+          {this.props.customHTML && (
```
Comment (gomezjuliana) [inline src/components/Banner/Banner.jsx]: What do you think of this @hamstu ??

File: src/components/Text/Text.jsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ Text.propTypes = {
   /** It's the name of the input it refers to. <br><i>This is only used for `label`.</i> */
   htmlFor: PropTypes.string,
   /** It change the color of the text. <br><i>This is only used for `label`</i>. */
-  light: PropTypes.bool,
+  color: PropTypes.string,
```
Comment (gomezjuliana) [inline src/components/Text/Text.jsx]: @federicoweber I changed this so we can customize the color.. is that ok with what you had in mind?

File: src/components/Banner/Banner.jsx (hunk 1 above)
Comment (hamstu) [reply src/components/Banner/Banner.jsx]: That's perfect @gomezjuliana!

File: src/components/Banner/Banner.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  text: PropTypes.string,
+
+  /** (Optional) The text of the Call To Action of the banner */
+  actionButton: PropTypes.shape({
+    label: PropTypes.string,
+    action: PropTypes.func,
+  }),
+
+  /** (Optional) custom html */
+  customHTML: PropTypes.shape({}),
```
Comment (hamstu) [inline src/components/Banner/Banner.jsx]: We could probably do something like this?

```suggestion
  customHTML: PropTypes.shape({ __html: PropTypes.string }),
```

File: src/components/AppShell/AppShell.jsx [hunk 4]
```diff
@@ -39,11 +57,21 @@ AppShell.propTypes = {
 
   /** (Optional) Your content component. */
   content: PropTypes.node.isRequired,
+
+  /** (Optional) Content of banner displayed below the navbar */
+  bannerOptions: PropTypes.shape({
+    text: PropTypes.string,
+    actionButton: PropTypes.shape({
+      label: PropTypes.string,
+      action: PropTypes.func,
```
Comment (hamstu) [inline src/components/AppShell/AppShell.jsx]: Might be good to add the `customHTML` prop here too for visibility?

**Pull request merged**